Restoring a checkpoint panics (in pod with infra container)

**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Restoring a checkpoint in a pod panics when the pod was created with infra container enabled (Default: true).

**Steps to reproduce the issue:**
1.  Run the [criu podman example](https://criu.org/Podman) in a pod.
```
root@debian10pod:~# podman pod create --name test
root@debian10pod:~# podman run -d --pod test busybox /bin/sh -c 'i=0; while true; do echo $i; i=$(expr $i + 1); sleep 1; done'
root@debian10pod:~# podman run -d --pod test busybox /bin/sh -c 'i=0; while true; do echo $i; i=$(expr $i + 1); sleep 1; done'
root@debian10pod:~# podman ps
CONTAINER ID  IMAGE                             COMMAND               CREATED         STATUS             PORTS   NAMES
e4764306e721  docker.io/library/busybox:latest  /bin/sh -c i=0; w...  10 seconds ago  Up 10 seconds ago          fervent_poincare
c9b1ef07799c  docker.io/library/busybox:latest  /bin/sh -c i=0; w...  12 seconds ago  Up 11 seconds ago          affectionate_nightingale
3b283845edd4  k8s.gcr.io/pause:3.2                                    22 seconds ago  Up 12 seconds ago          45b7863f398a-infra
```
2.
```
root@debian10pod:~# podman container checkpoint e4764306e721
e4764306e72125f7fd84a3c886904f07473009b072c3e553b9492cd7180f6c28
```
3.
```
root@debian10pod:~# podman container restore e4764306e721
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/containers/podman/libpod.(*Container).restore(0xc000230000, 0x1e4b160, 0xc000130020, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/src/packages/BUILD/src/github.com/containers/podman/libpod/container_internal_linux.go:990 +0x244a
github.com/containers/podman/libpod.(*Container).Restore(0xc000230000, 0x1e4b160, 0xc000130020, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/src/packages/BUILD/src/github.com/containers/podman/libpod/container_api.go:715 +0x174
github.com/containers/podman/pkg/domain/infra/abi.(*ContainerEngine).ContainerRestore(0xc000010c60, 0x1e4b160, 0xc000130020, 0xc00051e4d0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/src/packages/BUILD/src/github.com/containers/podman/pkg/domain/infra/abi/containers.go:541 +0x24e
github.com/containers/podman/cmd/podman/containers.restore(0x2be0000, 0xc00051e4d0, 0x1, 0x1, 0x0, 0x0)
        /usr/src/packages/BUILD/src/github.com/containers/podman/cmd/podman/containers/restore.go:88 +0x1ad
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).execute(0x2be0000, 0xc000136030, 0x1, 0x1, 0x2be0000, 0xc000136030)
        /usr/src/packages/BUILD/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:838 +0x453
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2bf4040, 0xc000130020, 0x18a4d60, 0x2c9b990)
        /usr/src/packages/BUILD/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:943 +0x317
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /usr/src/packages/BUILD/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:883
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /usr/src/packages/BUILD/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:876
main.Execute()
        /usr/src/packages/BUILD/src/github.com/containers/podman/cmd/podman/root.go:86 +0xec
main.main()
        /usr/src/packages/BUILD/src/github.com/containers/podman/cmd/podman/main.go:77 +0x18c
```

**Describe the results you received:**
Panics when restoring a container.

**Describe the results you expected:**
Should not panic. `podman pod create --name test --infra=false` and everything works perfectly fine.


**Output of `podman version`:**

```
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.14
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.20, commit: '
  cpus: 4
  distribution:
    distribution: debian
    version: "10"
  eventLogger: journald
  hostname: debian10pod
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.8.0-0.bpo.2-amd64
  linkmode: dynamic
  memFree: 3895685120
  memTotal: 4128940032
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.0.0~rc6+dfsg1
      commit: 1.0.0~rc6+dfsg1-3
      spec: 1.0.1
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 1m 29.7s
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 5
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.14
  OsArch: linux/amd64
  Version: 2.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**
```
root@debian10pod:~# apt list podman
Listing... Done
podman/unknown,now 2.1.1~2 amd64 [installed]
podman/unknown 2.1.1~2 arm64
podman/unknown 2.1.1~2 armhf
podman/unknown 2.1.1~2 ppc64el
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**
Yes

--- a/libpod/container_internal_linux.go
+++ b/libpod/container_internal_linux.go
@@ -1024,13 +1024,15 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
 		if !options.IgnoreStaticMAC {
 			// Take the first device with a defined sandbox.
 			var MAC net.HardwareAddr
-			for _, n := range networkStatus[0].Interfaces {
-				if n.Sandbox != "" {
-					MAC, err = net.ParseMAC(n.Mac)
-					if err != nil {
-						return errors.Wrapf(err, "failed to parse MAC %v", n.Mac)
+			if len(networkStatus) > 0 {
+				for _, n := range networkStatus[0].Interfaces {
+					if n.Sandbox != "" {
+						MAC, err = net.ParseMAC(n.Mac)
+						if err != nil {
+							return errors.Wrapf(err, "failed to parse MAC %v", n.Mac)
+						}
+						break
 					}
-					break
 				}
 			}
 			if MAC != nil {

